models/omnivore_model.py

```diff
@@ -42,26 +42,32 @@ def get_kinetics_head(dim_in: int = 1024, num_classes: int = 400) -> nn.Module:
 
 
 class OmnivoreModel(nn.Module):
-    def __init__(self, trunk: nn.Module, heads: nn.ModuleDict):
+    def __init__(self, trunk: nn.Module, heads: Union[nn.ModuleDict, nn.Module]):
         super().__init__()
         self.trunk = trunk
         self.heads = heads
         self.types = ["image", "video", "rgbd"]
-        assert all([n in heads for n in self.types]), "All heads must be provided"
+        self.base_model = False
+        if isinstance(heads, nn.ModuleDict):
+            self.base_model = True
+            assert all([n in heads for n in self.types]), "All heads must be provided"
 
-    def forward(self, x: torch.Tensor, input_type: str = "image"):
+    def forward(self, x: torch.Tensor, input_type: Optional[str] = "image"):
```

Comment:
The default can be made `None` - otherwise this looks confusing for single head video models. 

And for multimodal models if `input_type` is `None` we use `image` for BC?